api/envoy/config/route/v3/route_components.proto

```diff
@@ -2375,7 +2375,6 @@ message InternalRedirectPolicy {
 // :ref:`Route.typed_per_filter_config<envoy_v3_api_field_config.route.v3.Route.typed_per_filter_config>`,
 // or :ref:`WeightedCluster.ClusterWeight.typed_per_filter_config<envoy_v3_api_field_config.route.v3.WeightedCluster.ClusterWeight.typed_per_filter_config>`
 // to add additional flags to the filter.
-// [#not-implemented-hide:]
 message FilterConfig {
```

Comment:
Emm...a question about why we need `is_optional` for pre route config?

From the usecase of `is_optional`, if the user want to avoid the new filter break the old envoy, then it should mark the filter and its pre route config as optional.

Reply:
I think new config won't break envoy. It just result in a configuration rejection. (Although configuration rejection is dangerous at initialzing phase)

> From the usecase of is_optional, if the user want to avoid the new filter break the old envoy, then it should mark the filter and its pre route config as optional.

I think maybe https://github.com/envoyproxy/envoy/pull/27263 is the answer?
And may be the initial design is to mark these be optional  separatedly? Because there maybe a exist filter to extend itself's function, for example, add the route level config support.

Reply:
> I think new config won't break envoy. It just result in a configuration rejection. (Although configuration rejection is dangerous at initialzing phase)
> 

yes, it won't break envoy. I mean it will break the functionality of the node with old version envoy. The control plane pass down the new filter, then the old version envoy reject the config, then that node won't work without config.